code/modules/mob/living/living.dm

```diff
@@ -952,4 +952,17 @@
 
 //used in datum/reagents/reaction() proc
 /mob/living/proc/get_permeability_protection()
-	return 0
\ No newline at end of file
+	return 0
+
+/mob/living/verb/sniff()
+
+	set name = "Sniff"
+	set category = "IC"
+
+	if(istype(wear_mask, /obj/item/clothing/mask))//can't smell things with a mask on your face
+		src << "<span class='notice'>You can't smell anything other than your mask.</span>"
```

Comment:
If you want this to actually prevent the smell, add a `return` statement on the next line.